src/crud/message.py

```
@@ -347,3 +348,363 @@ async def update_message(
+
+        for match in matches:
+            start = match.seq_in_session - context_window
+            end = match.seq_in_session + context_window
+
+            if merged_ranges and start <= merged_ranges[-1][1] + 1:
+                # Overlaps with previous range - merge
+                prev_start, prev_end, prev_matches = merged_ranges[-1]
+                merged_ranges[-1] = (
+                    prev_start,
+                    max(prev_end, end),
+                    prev_matches + [match],
```

Ruff (0.14.10)

Consider `[*prev_matches, match]` instead of concatenation

Replace with `[*prev_matches, match]`

(RUF005)